src/dd_init.F90

```diff
@@ -0,0 +1,327 @@
+            ! Second-order neighbors (neighbors of the first-order neighbors)
+            ! -x
+      mesh_indices_to_bin(dd % mesh, dd % ijk + (/-2,  0,  0/)), & !-x
+            dd % meshbin                                             , & !+x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1, -1,  0/)), & !-y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  1,  0/)), & !+y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  0, -1/)), & !-z
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  0,  1/)), & !+z
+            ! +x
+      dd % meshbin                                             , & !-x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 2,  0,  0/)), & !+x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1, -1,  0/)), & !-y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  1,  0/)), & !+y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  0, -1/)), & !-z
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  0,  1/)), & !+z
+            ! -y
+      mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1, -1,  0/)), & !-x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1, -1,  0/)), & !+x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0, -2,  0/)), & !-y
+            dd % meshbin                                             , & !+y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0, -1, -1/)), & !-z
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0, -1,  1/)), & !+z
+            ! +y
+      mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  1,  0/)), & !-x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  1,  0/)), & !+x
+            dd % meshbin                                             , & !-y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  2,  0/)), & !+y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  1, -1/)), & !-z
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  1,  1/)), & !+z
+            ! -z
+      mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  0, -1/)), & !-x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  0, -1/)), & !+x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0, -1, -1/)), & !-y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  1, -1/)), & !+y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  0, -2/)), & !-z
+            dd % meshbin                                             , & !+z
+            ! +z
+      mesh_indices_to_bin(dd % mesh, dd % ijk + (/-1,  0,  1/)), & !-x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 1,  0,  1/)), & !+x
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0, -1,  1/)), & !-y
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  1,  1/)), & !+y
+            dd % meshbin                                             , & !-z
+            mesh_indices_to_bin(dd % mesh, dd % ijk + (/ 0,  0,  2/))  & !+z
+                 /)
```

Comment:
Could probably make this more short, clear, and less error-prone with a set of `do` loops.